provisioner/terraform/provision_test.go

```diff
@@ -165,8 +166,18 @@ func TestProvision_Cancel(t *testing.T) {
 
 			// Example: exec /path/to/terrafork_fake_cancel.sh 1.2.1 apply "$@"
 			content := fmt.Sprintf("#!/bin/sh\nexec %q %s %s \"$@\"\n", fakeBin, terraform.TerraformVersion.String(), tt.mode)
-			err := os.WriteFile(binPath, []byte(content), 0o755) //#nosec
+
+			// golang's standard OS library can sometimes leave the file descriptor open even after
+			// "Closing" the file (which can then lead to a "text file busy" error, so we bypass this
+			// and use syscall directly).
```

Comment:
Is this a known / documented issue in the stdlib?
Would be nice to remove this workaround if it gets fixed in a later version.